sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala

```diff
@@ -473,10 +536,35 @@ case class GpuFileSourceScanExec(
       QueryPlan.normalizePredicates(dataFilters, output),
       None)
   }
+
+  private def canUseSmallFileOptimization: Boolean = {
+    // get up to date conf because isInputFileExecUsed is set at runtime
+    val rapidsConf = new RapidsConf(relation.sparkSession.sessionState.conf)
+    val formatSupportsSmallFilesOptimization = relation.fileFormat match {
+      case _: ParquetFileFormat => true
+      case _ => false
+    }
+    logDebug(s"Small file optimization: ${rapidsConf.isParquetSmallFilesEnabled} " +
+      s"format: $formatSupportsSmallFilesOptimization " +
+      s"Inputfile: ${rapidsConf.isInputFileExecUsed}")
+
+    (rapidsConf.isParquetSmallFilesEnabled && formatSupportsSmallFilesOptimization
+      && !rapidsConf.isInputFileExecUsed)
+  }
+
 }
 
 object GpuFileSourceScanExec {
   def tagSupport(meta: SparkPlanMeta[FileSourceScanExec]): Unit = {
+    val sparkSession = meta.wrapped.sqlContext.sparkSession
+    val options = meta.wrapped.relation.options
+    if (meta.conf.isParquetSmallFilesEnabled &&
+      (sparkSession.conf.getOption("spark.sql.parquet.mergeSchema").exists(_.toBoolean) ||
+      options.getOrElse("mergeSchema", "false").toBoolean)) {
+
+      meta.willNotWorkOnGpu("mergeSchema is not supported yet with" +
```

Comment:
Shouldn't this check be within the `ParquetFileFormat` case below rather than independent of what file format we're trying to scan?

Reply:
this will move with changes to mergeSchema